handlers.go

```diff
@@ -62,3 +62,22 @@ func RecoverImmediately(ctx context.Context, dagst *DAGStore, failureCh chan Sha
 		}
 	}
 }
+
+// LogTraceLoop logs the output of the given trace channel until the given context expires.
+func LogTraceLoop(ctx context.Context, traceCh chan Trace, onDone func()) {
+	defer onDone()
```

Comment:
I'm not sure I understand why you need the onDone method here. Instead of passing this function as a parameter, can't the caller just do:
```
LogTraceLoop(ctx, ch)
onDone()
```

Reply:
Removed this function completely.